src/pages/ProductListPage/ProductListPage.js

```diff
@@ -0,0 +1,44 @@
+import './ProductListPage.css';
+import Sidebar from "../../components/Sidebar/Sidebar";
+import ProductsList from "../../components/ProductsList/ProductsList";
+import {results as data} from '../../utils/mocks/en-us/products.json'
+import Footer from "../../components/Footer/Footer";
+import { useState, useEffect} from "react";
+
+const ProductListPage = () => {
+  const [filteredProducts, setFilteredProducts] = useState(data);
+  const [activeCategories, setActiveCategories] = useState([]);
+
+  useEffect(() => {
+    window.scrollTo(0, 0)
+  }, [])
+
+  const handleOnClick = (categoryId) => {
+    if (activeCategories.includes(categoryId)) {
+      const index = activeCategories.indexOf(categoryId);
+      activeCategories.splice(index,1);
+      setActiveCategories([...activeCategories])
+    } else {
+      setActiveCategories([...activeCategories, categoryId]);
+    }
+  };
+
+  useEffect(() => {
+    const filtered = data.filter((product) =>  activeCategories.includes(product.data.category.id));
+    if (activeCategories.length > 0) {
+      setFilteredProducts(filtered);
+    } else {
+      setFilteredProducts(data);
+    }
+  }, [activeCategories]);
```

Comment:
Quick recommendation: just move the filtering inside the if, so we don't filter every time, just when needed.  
```suggestion
  useEffect(() => {
    if (activeCategories.length > 0) {
    const filtered = data.filter((product) =>  activeCategories.includes(product.data.category.id));
      setFilteredProducts(filtered);
    } else {
      setFilteredProducts(data);
    }
  }, [activeCategories]);
```